src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Connection/PreloginSubHandlers/PreloginPacketHandler.cs

```diff
@@ -0,0 +1,412 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.Common;
+using Microsoft.Data.SqlClient;
+using Microsoft.Data.SqlClientX.IO;
+
+namespace Microsoft.Data.SqlClientX.Handlers.Connection.PreloginSubHandlers
+{
+    internal class PreloginPacketHandler : IHandler<PreLoginHandlerContext>
+    {
+        const int GUID_SIZE = 16;
```

Comment:
Can we push all constants to a dedicated Constants.cs file?

Reply:
@cheenamalhotra, I want to stay away from these Global constants files, which almost always end up as an anti-pattern for separating code. 

In this case GUID_SIZE is only needed by PreloginPacketHandler. So I decided to scope it to this particular class.

Reply:
I meant generally, moving constants to shared file for readability, sometimes it's hard to locate constants otherwise. But not a critical feedback.

Reply:
@cheenamalhotra  Got it. Definitely. GUID_SIZE is only used in Prelogin, hence I am leaving it here.